pkg/KubeArmorOperator/internal/controller/resources.go

```diff
@@ -460,6 +460,7 @@ func (clusterWatcher *ClusterWatcher) WatchRequiredResources() {
 	relayServer.Spec.Template.Spec.Containers[0].ImagePullPolicy = corev1.PullPolicy(common.KubeArmorRelayImagePullPolicy)
 	deploys := []*appsv1.Deployment{
 		addOwnership(relayServer).(*appsv1.Deployment),
+		addOwnership(controller).(*appsv1.Deployment),
```

Comment:
let's not add controller deployment to `deploys` controller deployment need to handled separately here:
https://github.com/kubearmor/KubeArmor/blob/b86dbcc76f94a5520f717331f9e490091c32ebbe/pkg/KubeArmorOperator/internal/controller/resources.go#L347